File: domain-driven-design/nodejs-web/src/infrastructure/repository/repository.js
package repository

// Repository 通用仓储接口，定义了常见的 CRUD 操作
type Repository[T any] interface {
  Save(entity *T) error         // 保存实体
  FindByID(id int) (*T, error)  // 根据ID查找实体
  FindAll() ([]*T, error)       // 查找所有实体
  Delete(id int) error          // 删除实体
}

File: domain-driven-design/java-web/src/main/java/com/javaweborder/infrastructure/repository/Repository.java
package com.javaweborder.infrastructure.repository;

import java.util.List;
import java.util.Optional;

public interface Repository<T> {

    void save(T entity);              // 保存实体

    Optional<T> findById(long id);     // 根据 ID 查找实体

    List<T> findAll();                // 查找所有实体

    void delete(long id);              // 删除实体
}
